mpl_gui/__init__.py

```diff
@@ -219,20 +253,27 @@ def close_all(self):
         passing it to `show`.
 
         """
-        for fig in self.figures:
-            if fig.canvas.manager is not None:
-                fig.canvas.manager.destroy()
+        for fig in list(self.figures):
+            self.close(fig)
+
+    def close(self, val):
+        if val == "all":
+            return self.close_all()
+        # or do we want to close _all_ of the figures with a given label / number?
+        if isinstance(val, str):
+            fig = self.by_label[val]
+        elif isinstance(val, int):
+            fig = self.by_number[val]
+        else:
+            fig = val
+        if fig.canvas.manager is not None:
+            fig.canvas.manager.destroy()
             # disconnect figure from canvas
             fig.canvas.figure = None
             # disconnect canvas from figure
             _FigureCanvasBase(figure=fig)
```

Comment:
close should definitely get a docstring here.  What does plt.close(fig) actually mean?  Even reading the code I'm not sure.  Since this is a clean API, it might be good to be explicit, even if that means changing some pyplot behaviour?

Reply:
Added a docstring.